[CNV#69839] Enable IBM Secure Execution on ocpvirt (#99764)
openshift/openshift-docs: modules/virt-launching-ibm-secure-execution-vm-ibm-z.adoc

// Module included in the following assemblies:
//
// * virt/virtual_machines/creating_vm/virt-configuring-ibm-secure-execution-vms-ibm-z.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-launching-ibm-secure-execution-vm-ibm-z_{context}"]
= Launching an {ibm-title} Secure Execution VM on {ibm-z-title} and {ibm-linuxone-title}

> Before launching an {ibm-name} Secure Execution VM on {ibm-z-name} and {ibm-linuxone-name}, you must add the `launchSecurity` parameter to the `secure-execution-vm.yaml` file . Otherwise, the VM does not boot correctly because it does not have access to the devices.

Reviewer: Minor typo:

```suggestion
Before launching an {ibm-name} Secure Execution VM on {ibm-z-name} and {ibm-linuxone-name}, you must add the `launchSecurity` parameter to the `secure-execution-vm.yaml` file. Otherwise, the VM does not boot correctly because it does not have access to the devices.
```

Also, is this file always named `secure-execution-vm.yaml` or should this be generically referred to as "you must add the `launchSecurity` parameter to the VM manifest" or similar?

Reviewer 2: it is not a must to name it always "secure-execution-vm.yaml" , it can be anyname.yaml so "to the VM manifest" is correct

.Procedure

. Apply the following machine configuration:
+
[source,yaml]
----
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  labels:
    kubevirt.io/vm: f41-se
  name: f41-se
spec:
  runStrategy: Always
  template:
    metadata:
      labels:
        kubevirt.io/vm: f41-se
    spec:
      domain:
        launchSecurity: {}
        devices:
          disks:
          - disk:
              bus: virtio
            name: rootfs
        machine:
          type: ""
        resources:
          requests:
            memory: 4Gi
      terminationGracePeriodSeconds: 0
      volumes:
        - name: rootfs
          dataVolume:

>             name: f41-se

Reviewer: This is a very specific VM manifest; is it always these exact parameters? If not, please update it (using the description list format) so that it is clear which parameters need to be changed by the customer.

----

[NOTE]
====

> As the memory of the VM is protected, {ibm-name} Secure Execution VMs are not live migratable. The VMs
> can only be migrated offline.

Reviewer: ISG nit; also removing hard wrap.

```suggestion
Because the memory of the VM is protected, {ibm-name} Secure Execution VMs are not live migratable. The VMs can only be migrated offline.
```

<img width="285" height="53" alt="Image" src="https://github.com/user-attachments/assets/7bff66d5-95d4-4580-8f5f-163dc1deb0d2" />

// Module included in the following assemblies:
//
// * virt/virtual_machines/creating_vm/virt-configuring-ibm-secure-execution-vms-ibm-z.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-launching-ibm-secure-execution-vm-ibm-z_{context}"]
= Launching an {ibm-title} Secure Execution VM on {ibm-z-title} and {ibm-linuxone-title}

Before launching an {ibm-name} Secure Execution VM on {ibm-z-name} and {ibm-linuxone-name}, you must add the `launchSecurity` parameter to the VM manifest. Otherwise, the VM does not boot correctly because it does not have access to the devices.

.Procedure

> . Apply the following `VirtualMachine` manifest to the cluster:

Reviewer: If you're keeping this as one step, please change the `.` to `*`

+
[source,yaml]
----
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  labels:
    kubevirt.io/vm: f41-se
  name: f41-se
spec:
  runStrategy: Always
  template:
    metadata:
      labels:
        kubevirt.io/vm: f41-se
    spec:
      domain:
        launchSecurity: {}
        devices:
          disks:
          - disk:
              bus: virtio
            name: rootfs
        machine:
          type: ""
        resources:
          requests:
            memory: 4Gi
      terminationGracePeriodSeconds: 0
      volumes:
        - name: rootfs
          dataVolume:
            name: f41-se
----
+

> where:
> +
> [source,yaml]
> ----
> spec:
>    domain:
>      launchSecurity:
> ----
> +
> Is required to launch  {ibm-name} Secure Execution VMs, the other values can vary depending on the setup.

Reviewer: ```suggestion
To launch {ibm-name} Secure Execution VMs, you must include the following YAML in the manifest:
+
[source,yaml]
----
spec:
   domain:
     launchSecurity:
----
+
The rest of the VM manifest is variable depending on your setup.
```
